CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.13.4)
+project(CMakePublic)
+
+add_library(CMakePublic cmake_public.cpp)
+
+#include header file
+target_include_directories (CMakePublic PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+message("Processing CMakeLists.txt")
```

Comment:
From what I was able to glean from the internet the two "installs" I added are sufficient to install the library and create the config file -- still a little fuzzy on the purpose of the file and how it accomplishes its goals but my current understanding is that when downstream dependencies call find_packages CMake will look for the config file and it will provide it important information on the library. Still need to do some more testing/research on this but here's the draft.

Reply:
I think what you have is fine for this project (subject to my suggested changes); as the projects get more involved the auto-generated config file likely won't cut it anymore (the reason why we have CMakePublicDepend2 is because it can be tricky to make sure that all of the information required to find CMakePublic is stored in the config file of CMakePublicDepend) . 

Part of the reason we're going through this exercise is because we want CMaize to automate this whole process to the extent possible, which requires knowing the general ways of doing all this stuff. Unfortunately, as you're likely learning, the existing documentation really sucks and cuts a lot of corners (like hard-coding paths etc.). It's also my hope that these repos can serve as a better source of documentation in the future, which is why I'm trying to do everything "right" in them (right in CMake is a moving target though...).